src/other/tables.jl

```diff
@@ -0,0 +1,32 @@
+using Tables, IteratorInterfaceExtensions
+
+Tables.schema(df::DataFrame) = NamedTuple{Tuple(names(df)), Tuple{eltypes(df)...}}
+Tables.AccessStyle(df::DataFrame) = Tables.ColumnAccess()
+Tables.istable(::Type{DataFrame}) = true
+Tables.rows(df::DataFrame) = eachrow(df)
+Tables.columns(df::DataFrame) = df
+
+function _isiterabletable(x::T) where {T}
+    Base.isiterable(T) || return false
+    return Base.IteratorEltype(x) === Base.HasEltype() ? eltype(x) <: NamedTuple : false
+end
+
+function DataFrame(x::T) where {T}
+    if Tables.istable(T)
+        return DataFrame([collect(u) for u in Tables.columns(x)],
+                          collect(Tables.names(Tables.schema(x))))
+    end
+    y = IteratorInterfaceExtensions.getiterator(x)
+    if _isiterabletable(y)
+        return DataFrame([collect(u) for u in Tables.buildcolumns(eltype(y), y)],
+                          collect(Tables.names(eltype(y))))
+    else
+        convert(DataFrame, x)
```

Comment:
I guess Base moved out from default fallback of constructors to `convert`. I do not know what would be best - trying to `convert` or throw an error immediately.

If we want to go with trying a conversion then in particular in general `convert` could create something else than a `DataFrame` so maybe it would be cleaner to return `x` here and above add type assertion:
```
function DataFrame(x::T)::DataFrame where {T}
```

Reply:
`convert(DataFrame, x)` must always return a `DataFrame`, but I'm not sure why we would call `convert` from the constructor, since Base has removed that fallback. Types should define their custom constructors if needed.

Reply:
Yes, but people aren't supposed to do that so we don't care what happens if they do. :-)

Reply:
I agree - but I just want to say, that if we want to allow it it is better to do return type assertion I think.

But as you have said - maybe we should error then and not try doing conversion.